### PR TITLE
Make dependency on re2 explicit for grpc, py3-grpcio, onnxruntime

### PR DESCRIPTION
**Pull request by iarspider**
Do not merge yet, running a local build

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test for el8_ppc64le_gcc10

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc10

**Comment by smuzaffar** [comment]
please test for el8_amd64_gcc11

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
Let's test for one arch first.

**Comment by iarspider** [comment]
please test

File: triton-inference-client.spec [hunk 1]
```diff
@@ -90,7 +90,7 @@ cmake ${PROJ_DIR} \
     -DTRITON_VERSION=%{realversion} \
     -DCMAKE_CXX_FLAGS="-Wno-error -fPIC" \
     -DFETCHCONTENT_SOURCE_DIR_REPO-COMMON=${COMMON_DIR} \
-    -DCMAKE_PREFIX_PATH="${GRPC_ROOT};${ABSEIL_CPP_ROOT}"
+    -DCMAKE_PREFIX_PATH="${GRPC_ROOT};${ABSEIL_CPP_ROOT};${RE2_ROOT}"
```
Comment (smuzaffar) [inline triton-inference-client.spec:93]: @iarspider , please also add an explicit dependency on `re2` here

Comment (iarspider) [reply triton-inference-client.spec:93]: One more reason to move to Spack: automatic propagation of dependencies :)

**Comment by iarspider** [comment]
please test

File: triton-inference-client.spec (hunk 1 above)
Comment (iarspider) [reply triton-inference-client.spec:93]: Do I need to add re2 to toolfile as well?

**Comment by smuzaffar** [comment]
please test

File: triton-inference-client.spec (hunk 1 above)
Comment (smuzaffar) [reply triton-inference-client.spec:93]: yes

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
@smuzaffar the comparision looks fine now (only two failures compare to 24, with more compared histograms - 3618332 vs 3618326)

**Comment by smuzaffar** [comment]
+externals

**Pull request merged**